### DISCUSSION TITLE
Airflow scheduler un-stable behaviour

### DISCUSSION THREAD

**Discussion by ilhnctn** [apache/airflow#15376] [Q&A]
Hi all,
I'm using [airflow-helm](https://github.com/airflow-helm/charts/tree/main/charts/airflow)'s helm chart to deploy Airflow on GCP Kubernetes environment.

Systems mostly seems to work properly but scheduler randomly goes off and stays off for too long times (sometimes more than two hours).
Since the pod is up & running healthcheck doesn't require a restart of pod but actually scheduler service is not active in the meanwhile.

Technical details are listed in the [issue](https://github.com/airflow-helm/charts/issues/138) on chart's repo.

Any idea/recommendation is appreciated.

Thanks in advance!

**Comment by ashb** [comment]
If you log in to the UI do see a banner about "scheduler is not heartbeating"?

How many dag files do you have, and do you have and DAGs with define callbacks (`on_sucess_callback` or similar) or use SLAs?

**Reply by ilhnctn** [reply]
Yes, i see it most of the times.
I have 24 DAG files (in different folders) and none of them have `on_sucess_callback` or any other callback.
But they all have slas. All SLAs are between 5 - 14 minutes.

How can this affect (or stop the service)? @ashb

**Reply by ashb** [reply]
Can you Airflow try 2.0.2?

**Reply by ilhnctn** [reply]
Need to check. I'm using 2.0.1 right now

**Reply by ilhnctn** [reply]
Apparently it cant install (helm gives a timeout). The only change i made is as below. Neither helm upgrade, nor helm install didn't succeed. 

```yaml
airflow:
  image:
    tag: 2.0.2
    ```

**Reply by ilhnctn** [reply]
It works now, will monitor for a few hours and write here again.

**Reply by ilhnctn** [reply]
In approximately 1.5 hours of monitoring, it seems like this time it comes back sooner. The longest unavailable period was 767 seconds so far, it comes and goes :)
If i could somehow reduce the offline time under 5 minutes we would use in production.

**Reply by ashb** [reply]
When it is not working do you see a yellow banner in the UI about "scheduler is not heartbeating"?

**Reply by ilhnctn** [reply]
Yes. Currently it is as below
```text
The scheduler does not appear to be running. Last heartbeat was received 22 minutes ago.

The DAGs list may not update, and new tasks will not be scheduled.
```

**Reply by ilhnctn** [reply]
The annoying thing is, on pods & services everything is stable & running. CPU and Memory consumption are even less than %30 of cluster (in total).
